src/Core/Types/Configuration/TypeInitializer.cs

```diff
@@ -324,8 +332,13 @@ private void RegisterExternalResolvers(DiscoveredTypes discoveredTypes)
             {
                 ParameterInfo parent = m.GetParameters()
                     .FirstOrDefault(t => t.IsDefined(typeof(ParentAttribute)));
-                return parent == null
-                    || parent.ParameterType.IsAssignableFrom(sourceType);
+                ParentAttribute attribute
```

Comment:
What does this do? Why do we need to know this detail here?

The parent is important where we collect the fields but should not have any concern if it is compile relevant.

Reply:
In case  of Include<TResolver> or the resolver of attribute we do not know the parent type. It would not necessarily be inferred correctly. There can be resolvers of different types in one class

Reply:
That is an incorrect assumption. There is always only one source type. Even when the type is composed of multiple types.